HACKING.md

```diff
@@ -0,0 +1,409 @@
+# Developing Salt
+
```

Comment:
**Praise:** So far, I'm loving this, Wayne. A huge improvement and a huge step forward. If you don't want to adopt my suggestions, we could spin them off into a separate issue and address them later.